proof/crefine/AARCH64/Fastpath_C.thy

```diff
@@ -1052,25 +1047,15 @@ lemma ccorres_pre_getCTE2:
                                           \<longrightarrow> s \<in> P' cte} hs
          (getCTE p >>= (\<lambda>rv. f rv)) c"
   apply (rule ccorres_guard_imp2, erule ccorres_pre_getCTE)
-  apply (clarsimp simp: map_comp_Some_iff ccte_relation_def
-                        c_valid_cte_def cl_valid_cte_def
+  apply (clarsimp simp: map_comp_Some_iff ccte_relation_def c_valid_cte_def cl_valid_cte_def
                         c_valid_cap_def)
   done
 
-declare empty_fail_assertE[iff]
-
 declare empty_fail_resolveAddressBits[iff]
 
-lemma lookupExtraCaps_null:
-  "msgExtraCaps info = 0 \<Longrightarrow>
-     lookupExtraCaps thread buffer info = returnOk []"
-  by (clarsimp simp: lookupExtraCaps_def
-                     getExtraCPtrs_def liftE_bindE
-                     upto_enum_step_def mapM_Nil
-              split: Types_H.message_info.split option.split)
-
+(* this condition is copied directly from fastpath_mi_check_spec, which comes out of guards there *)
 lemma fastpath_mi_check:
-  "((mi && mask 9) + 3) && ~~ mask 3 = 0
+  "(\<not> 4 < mi && 0x1FF)
```

Comment:
maybe add a FIXME on this one for removing the magic numbers

Reply:
I put a comment right above which says this is copied by whatever shows up in `fastpath_mi_check_spec`, which I think are bitfield gen numbers? I have no clue what they are.